Extension/src/LanguageServer/settings.ts

```diff
@@ -414,337 +517,157 @@ export class CppSettings extends Settings {
+    public get defaultLimitSymbolsToIncludedHeaders(): boolean | undefined { return this.getAsBoolean("default.browse.limitSymbolsToIncludedHeaders"); }
+    public get defaultSystemIncludePath(): string[] | undefined { return this.getAsArrayOfStrings("default.systemIncludePath"); }
+    public get defaultEnableConfigurationSquiggles(): boolean | undefined { return this.getAsBoolean("default.enableConfigurationSquiggles"); }
+    public get defaultCustomConfigurationVariables(): { [key: string]: string } | undefined {return this.getAsKeyValueObject("default.customConfigurationVariables", "string", "string");}
+    public get useBacktickCommandSubstitution(): boolean | undefined { return this.getAsBoolean("debugger.useBacktickCommandSubstitution"); }
+    public get codeFolding(): boolean {return this.getAsString("codeFolding.enabled") === "enabled"; }
+    public get caseSensitiveFileSupport(): boolean { return !isWindows || this.getAsString("caseSensitiveFileSupport") === "enabled"; }
+    public get doxygenSectionTags(): string[] | undefined { return this.getAsArrayOfStrings("doxygen.sectionTags"); }
+    public get hover(): string | undefined { return this.getAsString("hover"); }
+    public get markdownInComments(): string | undefined { return this.getAsString("markdownInComments"); }
+    public get legacyCompilerArgsBehavior(): boolean | undefined { return this.getAsBoolean("legacyCompilerArgsBehavior"); }
+
+    public get inlayHintsAutoDeclarationTypes(): boolean { return this.getAsBoolean("inlayHints.autoDeclarationTypes.enabled"); }
+    public get inlayHintsAutoDeclarationTypesShowOnLeft(): boolean { return this.getAsBoolean("inlayHints.autoDeclarationTypes.showOnLeft"); }
+    public get inlayHintsParameterNames(): boolean { return this.getAsBoolean("inlayHints.parameterNames.enabled"); }
+    public get inlayHintsParameterNamesSuppressName(): boolean { return this.getAsBoolean("inlayHints.parameterNames.suppressWhenArgumentContainsName"); }
+    public get inlayHintsParameterNamesHideLeadingUnderscores(): boolean { return this.getAsBoolean("inlayHints.parameterNames.hideLeadingUnderscores"); }
+    public get inlayHintsReferenceOperator(): boolean { return this.getAsBoolean("inlayHints.referenceOperator.enabled"); }
+    public get inlayHintsReferenceOperatorShowSpace(): boolean { return this.getAsBoolean("inlayHints.referenceOperator.showSpace"); }
+    public get enhancedColorization(): boolean { return this.getAsString("enhancedColorization") === "enabled" && this.intelliSenseEngine === "default" && vscode.workspace.getConfiguration("workbench").get<string>("colorTheme") !== "Default High Contrast"; }
+    public get formattingEngine(): string | undefined { return this.getAsString("formatting"); }
+    public get vcFormatIndentBraces(): boolean { return this.getAsBoolean("vcFormat.indent.braces"); }
     public get vcFormatIndentMultiLineRelativeTo(): string {
```

Comment:
It turns out that the value returned by this getter can actually be written out to `.editorConfig`, so if `this.getAsString` will automatically lower-case the enum values, the value will be incorrect in the file we write this to. So this means that we can't automatically lower-case the strings in `this.getAsString`.

Reply:
Weren't we lowercasing these values before? Was this an existing bug? We currently process enum values as lowercase on the native side so I am not completely sure if it is necessary to make them lowercase in the typescript. Should we just disregard the lowercasing altogether?

Reply:
> Weren't we lowercasing these values before? 

no. Only for comparisons.

> Was this an existing bug? 

no. The setting returned the original value, but there could have been a bug if the user typed in an invalid value and we'd write that to editorConfig without any checks.  With this PR it should work as expected.

Reply:
So we should not be lowercasing the value after validation as we were doing before? (see below):
 
`public get intelliSenseEngineFallback(): boolean | undefined { return super.Section.get<string ("intelliSenseEngineFallback")?.toLowerCase() === "enabled"; }`

Reply:
you will need to lower-case it for comparisons.  This is the correct code:

```ts
public get intelliSenseEngineFallback(): string {
    return this.getAsString("intelliSenseEngineFallback");
}
public get isIntelliSenseEngineFallbackEnabled(): boolean {
    return this.intelliSenseEngineFallback.toLowerCase() === "enabled";
}
```